scio-extra/src/main/scala/com/spotify/scio/extra/transforms/package.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.extra
+
+import java.io.File
+import java.net.URI
+import java.nio.file.Path
+
+import com.spotify.scio.util.Functions
+import com.spotify.scio.values.SCollection
+import org.apache.beam.sdk.transforms.ParDo
+
+import scala.collection.JavaConverters._
+import scala.reflect.ClassTag
+
+/**
+ * Main package for transforms APIs. Import all.
+ */
+package object transforms {
+
+  /**
+   * Enhanced version of [[SCollection]] with [[URI]] methods.
+   */
+  implicit class URISCollection(val self: SCollection[URI]) extends AnyVal {
+    /**
+     * Download [[URI]] elements and process as local [[Path]]s.
+     * @param batchSize batch size when downloading files
+     * @param keep keep downloaded files after processing
+     */
+    def mapFile[T: ClassTag](f: Path => T,
+                             batchSize: Int = 10,
+                             keep: Boolean = false): SCollection[T] =
+      self.applyTransform(ParDo.of(new FileDoFn[T](Functions.serializableFn(f), batchSize, keep)))
+
+    /**
+     * Download [[URI]] elements and process as local [[Path]]s.
+     * @param batchSize batch size when downloading files
+     * @param keep keep downloaded files after processing
+     */
+    def flatMapFile[T: ClassTag](f: Path => TraversableOnce[T],
+                                 batchSize: Int = 10,
+                                 keep: Boolean = false): SCollection[T] =
+      self
+        .applyTransform(ParDo.of(new FileDoFn[TraversableOnce[T]](
+          Functions.serializableFn(f), batchSize, keep)))
+        .flatMap(identity)
+
+  }
+
+  /**
+   * Enhanced version of [[SCollection]] with pipe methods.
+   */
+  implicit class PipeSCollection(val self: SCollection[String]) extends AnyVal {
+
+    /**
+     * Pipe elements through an external command via StdIn & StdOut.
+     * @param command the command to call
+     * @param environment environment variables
+     * @param dir the working directory of the sub-process
+     */
+    def pipe(command: String,
```

Comment:
i wanna defer this for now. We could have a dummy transform (that doesn't process elements) like `p.runOnWorkers("apt-get install ...")` or `p.runOnWorkers("wget ...")` but that doesn't feel super natural.